spec/PeriodicBackgroundSync-index.bs

```diff
@@ -133,43 +150,47 @@ The user agent should cap the duration and frequency of these events to limit re
+    Note: User agents MAY also choose to update this value in response to changes in user engagement levels.
+  
+  <div algorithm>
+    Calculating the [=periodic sync registration/time to fire=] for |registration| involves running these steps:
+    1. If [=periodic sync registration/count of retries=] for |registration| is 0, <dfn>calculate the time to fire of the first attempt of</dfn> the [=periodicsync event=]:
+        1. Let |now|, a timestamp, represent the current time.
+        1. Let |origin| be the [=origin=] associated with |registration|'s [=periodic sync registration/service worker registration=].
+        1. Let |timeToFire| be |now|.
+        1. Increment |timeToFire| by the greater of [=minimum periodic sync interval for any origin=] and |registration|'s [=periodic sync registration/minimum interval=].
+        1. If [=periodic sync scheduler/time till the next scheduled periodicsync event=] for |origin| is not null, increment |timeToFire| such that the delta between |timeToFire| and [=periodic sync scheduler/time till the next scheduled periodicsync event=] for |origin| conforms to the cap enforced by [=minimum periodic sync interval for any origin=].
+        1. Increment |timeToFire| by some user agent defined amount. Note: User agents can use this to delay synchronization due to the user's engagement with |origin|.
+    1. Else:
+        1. If |registration|'s [=periodic sync registration/count of retries=] is greater than the [=maximum number of retries=] allowed, set [=periodic sync registration/count of retries=] to 0 and follow the steps to [=calculate the time to fire of the first attempt of=] the periodicsync event.
+        1. Else, Set |registration|'s [=periodic sync registration/time to fire=] to |now| + a small back-off that is exponentially proportional to [=periodic sync registration/count of retries=].
+  </div>
+
+  <div algorithm>
+    ## <dfn>Process periodic sync registrations</dfn> ## {#process-periodic-sync-registrations-algorithm}
+    1. While true:
+        1. Wait for [=minimum periodic sync interval across origins=].
+        1. Wait a user agent defined time. Note: This can used to group synchronization for different [=periodic sync registrations=] into a single device wake-up.
+        1. Wait until online.
+        1. For each [=/service worker registration=] |registration|, [=enqueue the following steps=] to its [=periodic sync processing queue=]:
+          1. For each [=periodic sync registration=] |periodicSyncRegistration| in |registration|'s [=active periodic sync registrations=]:
+            1. In Parallel: If |periodicSyncRegistration|'s [=periodic sync registration/time to fire=] is in the current time or in the past, [=fire a periodicsync event=] for |periodicSyncRegistration|.
```

Comment:
Link "in parallel"

Reply:
Also, you only want to do this for registrations that are "pending".

Reply:
done.